lib/widgets/app.dart

```diff
@@ -149,12 +149,23 @@ class MessageListPage extends StatelessWidget {
   Widget build(BuildContext context) {
     return Scaffold(
         appBar: AppBar(title: const Text("All messages")),
-        body: Center(
-            child: Column(children: const [
-          Expanded(child: MessageList()),
-          SizedBox(
-              height: 80,
-              child: Center(child: Text("(Compose box goes here.)"))),
-        ])));
+
+        // See MediaQuery.removePadding for why we need this BuildContext.
+        body: Builder(
+            builder: (BuildContext context) => Center(
+                child: Column(children: [
+                  MediaQuery.removePadding(
+                      // Scaffold knows about the app bar, and so has run this
+                      // BuildContext, which is under `body`, through
+                      // MediaQuery.removePadding with `removeTop: true`.
+                      context: context,
+
+                      // The compose box pads the bottom inset.
+                      removeBottom: true,
+
+                      child: const Expanded(child: MessageList())),
```

Comment:
The MessageList gets a bit lost here for me, visually — that is, it's easy to miss when looking for the main payload of this widget tree. Let's give it its own line:
```suggestion
                      child: const Expanded(
                          child: MessageList())),
```